### PR TITLE
refactor: move `getCurrentChainId` from `selectors/selectors.js` to `shared/modules/selectors/networks.ts`

### PR DESCRIPTION
**Pull request by davidmurdoch**
Converts the selector `getCurrentChainId` from functions from JS to TS and moves it to the `shared` folder. Also updates some functions to match the actual expect return values and fixes some types.

Why only this function? I'm trying to solve circular dependency issues. `getCurrentChainId` is so widely used in the codebase, it makes it very complicated to untangle. I've moved it to `shared/modules/selectors/networks.ts`

### CODE REVIEW

File: ui/selectors/networks.ts [hunk 1]
```diff
@@ -81,13 +81,13 @@ export const getProviderConfig = createSelector(
         }
       }
     }
-    return undefined; // should not be reachable
```
Comment (davidmurdoch) [inline ui/selectors/networks.ts:84]: One of two logic changes.

File: ui/selectors/institutional/selectors.ts [hunk 2]
```diff
@@ -172,12 +172,13 @@ export function getIsCustodianSupportedChain(
...
     const accountType = getAccountType(state);
-    const providerConfig = getProviderConfig(state);
 
-    if (!selectedAccount || !accountType || !providerConfig) {
+    if (!selectedAccount || !accountType) {
       throw new Error('Invalid state');
     }
 
+    const providerConfig = getProviderConfig(state);
+
```
Comment (davidmurdoch) [inline ui/selectors/institutional/selectors.ts:181]: One of two logic changes.

`getProviderConfig` now throws on its own if there isn't a `providerConfig`. This is teeccchnniiicaallly not supposed to be possible. See the implementation and changes to understand more.

File: shared/modules/selectors/networks.ts [hunk 3]
```diff
@@ -81,13 +81,13 @@ export const getProviderConfig = createSelector(
         }
       }
     }
-    return undefined; // should not be reachable
+    throw new Error('Provider configuration not found');
```
Comment (davidmurdoch) [inline shared/modules/selectors/networks.ts:84]: One of two logci changes: if this _is_ reachable, we should yell about it so we know there is a bug.

**Review by HowardBraham — DISMISSED** [review]

File: shared/modules/selectors/smart-transactions.ts [hunk 4]
```diff
@@ -108,7 +107,7 @@ export const getSmartTransactionsPreferenceEnabled = createSelector(
 );
 
 export const getCurrentChainSupportsSmartTransactions = (
-  state: SmartTransactionsMetaMaskState,
+  state: NetworkState,
```
Comment (davidmurdoch) [inline shared/modules/selectors/smart-transactions.ts:110]: Previous type wasn't enough now that `getCurrentChainId` is typed.

File: ui/hooks/ramps/useRamps/useRamps.ts [hunk 5]
```diff
@@ -32,7 +32,7 @@ const useRamps = (
   const isMarketingEnabled = useSelector(getDataCollectionForMarketing);
 
   const getBuyURI = useCallback(
-    (_chainId: ChainId | CaipChainId) => {
+    (_chainId: Hex | CaipChainId) => {
```
Comment (davidmurdoch) [inline ui/hooks/ramps/useRamps/useRamps.ts:35]: The caller of this function uses the result of calling `getCurrentChainId`, which now has a type, and  `ChainId | CaipChainId` is not compatible.

File: ui/selectors/selectors.js [hunk 6]
```diff
@@ -132,11 +133,6 @@ export function getNetworkIdentifier(state) {
   return nickname || rpcUrl || type;
 }
 
-export function getCurrentChainId(state) {
```
Comment (davidmurdoch) [inline ui/selectors/selectors.js:135]: this is the function that this PR moves.

**Review by danjm — DISMISSED** [review]
LGTM

**Review by HowardBraham — DISMISSED** [review]

**Review by jclancy93 — DISMISSED** [review]

**Review by GuillaumeRx — DISMISSED** [review]

**Review by NidhiKJha — DISMISSED** [review]

**Review by shane-t — DISMISSED** [review]

**Review by FrederikBolding — DISMISSED** [review]

**Review by HowardBraham — APPROVED** [review]

**Review by jclancy93 — APPROVED** [review]

**Review by shane-t — APPROVED** [review]

**Pull request merged**